src/test/java/seedu/address/logic/commands/DietCommandTest.java

```diff
@@ -0,0 +1,128 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_DIET_AMY;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_DIET_BOB;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.logic.commands.CommandTestUtil.showPetAtIndex;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PET;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_PET;
+import static seedu.address.testutil.TypicalPets.getTypicalAddressBook;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.model.AddressBook;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.pet.Diet;
+import seedu.address.model.pet.Pet;
+import seedu.address.testutil.PetBuilder;
+
+
+
+class DietCommandTest {
+
+    private static final String REMARK_STUB = "Some remark";
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_addDietUnfilteredList_success() {
+        Pet firstPerson = model.getFilteredPetList().get(INDEX_FIRST_PET.getZeroBased());
+        Pet editedPerson = new PetBuilder(firstPerson).withDiet(REMARK_STUB).build();
```

Comment:
```suggestion
        Pet firstPet = model.getFilteredPetList().get(INDEX_FIRST_PET.getZeroBased());
        Pet editedPet = new PetBuilder(firstPet).withDiet(DIET_STUB).build();
```
just minor naming changes, can find and replace all 'person' with 'pet' and 'REMARK' with 'DIET' below too.